tests/test_json.py

```diff
@@ -57,7 +58,7 @@ def test_device(mod_device):
 
 def test_virtual_device(mod_device):
     assert encode_decode(MockDevice) == MockDevice
-    virtual_mod = mod_device.to_virtual()
+    virtual_mod = replace(mod_device.to_virtual(), dmm_objects=())
```

Comment:
Is this just because we don't support the legacy serialization yet? It should be an xfail instead then

Reply:
I fixed it in my PR